### PR TITLE
Sharepoint integration

### PR DESCRIPTION
**Pull request by jbarbaFullScale**
- Added Option Sharepoint
- Refactored `cloudFileBrowser.init` function

### CODE REVIEW

File: divvy/cloudfilebrowser.js [hunk 1]
```diff
@@ -1015,10 +1025,14 @@ var cloudFileBrowser = (function() {
...
             if (element === 'sharepoint') {
             	siteAddress = $(container).find('#sp-site-address').val();
+            	apiKey = $(container).find('#client_key').val();
+            	apiSecret = $(container).find('#secret_key').val();
             }
             var callbackArgs = {
                 'element' : element,
-                'siteAddress': siteAddress
+                'siteAddress': siteAddress,
+                apiKey,
```
Comment (ganiserb) [inline divvy/cloudfilebrowser.js]: Is this valid JS? I thought this bit of syntactic sugar was ES6 only... cc @letoss you know more JS than I do.

File: divvy/cloudfilebrowser.js [hunk 2]
```diff
@@ -851,17 +856,22 @@ var cloudFileBrowser = (function() {
...
                     var element_id = '';
+                    var token_elements = '';
+                    var style = '';
                     if (serviceName === 'sharepoint') {
                         text = 'Share Point Site Addresss';
                         spacing = '</br>';
                         element_id = 'sp-';
+                        token_elements = '<div> <input type="text" id="client_key" placeholder="Client API Key"></div>';
+                        token_elements += '<div> <input type="text" id="secret_key" placeholder="API Secret Key"></div>';
+                        style = 'style="margin-top: 110px"';
```
Comment (ganiserb) [inline divvy/cloudfilebrowser.js:872]: I hate that we're mixing JS and HTML like this...

**Review by ganiserb — COMMENTED** [review]
Just a couple of questions...

File: divvy/cloudfilebrowser.js (hunk 1 above)
Comment (jbarbaFullScale) [reply divvy/cloudfilebrowser.js]: @letoss  please let me know what you think, I personally think that this should just be 
callbackArgs = {element, siteAddress, apiKey}

Comment (letoss) [reply divvy/cloudfilebrowser.js]: This is an ES6 syntax sugar as Gaby said. If this file belongs to the old FE, it may fail if it's run on IE11 or Edge.

Comment (nmallare) [reply divvy/cloudfilebrowser.js]: Reminder that we aren't supporting IE11 any more.

**Pull request closed without merging**